internal/xds/rbac/matchers.go

```diff
@@ -183,9 +183,14 @@ func matchersFromPrincipals(principals []*v3rbacpb.Principal) ([]matcher, error)
 			// The source ip principal identifier is deprecated. Thus, a
 			// principal typed as a source ip in the identifier will be a no-op.
 			// The config should use DirectRemoteIp instead.
-		case *v3rbacpb.Principal_RemoteIp: // Allow equating RBAC's direct_remote_ip...do we need this?
-			// Not supported in gRPC RBAC currently - a principal typed as
-			// Remote Ip in the initial config will be a no-op.
+		case *v3rbacpb.Principal_RemoteIp:
+			// RBAC in gRPC treats direct_remote_ip and remote_ip as logically
+			// equivalent, as per A41.
+			m, err := newSourceIPMatcher(principal.GetRemoteIp())
```

Comment:
Why is this called `newSourceIPMatcher` not just `newRemoteIPMatcher`?
Is it because this is server side, so `remote` is `source`?

Reply:
Good question, it's ugly (see diff in my tests for this PR - I had same thought) switched.